src/github/SyftGithubAction.ts

```diff
@@ -356,6 +352,44 @@ export async function runSyftAction(): Promise<void> {
   }
 }
 
+/**
+ * Attaches the SBOM assets to a release if run in release mode
+ */
+export async function uploadDependencySnapshot(): Promise<void> {
+  if (!uploadToSnapshotAPI()) {
+    return;
+  }
+
+  if (!fs.existsSync(githubDependencySnapshotFile)) {
+    core.warning(
+      `No dependency snapshot found at '${githubDependencySnapshotFile}'`
+    );
+    return;
+  }
+  const { job, runId, repo, sha, ref } = github.context;
+  const client = getClient(repo, core.getInput("github-token"));
+
+  const snapshot = JSON.parse(
+    fs.readFileSync(githubDependencySnapshotFile).toString("utf8")
+  ) as DependencySnapshot;
+
+  // Need to add the job and repo details
+  snapshot.job = {
+    name: job,
+    id: `${runId}`,
+  };
+  snapshot.sha = sha;
+  snapshot.ref = ref;
+
+  core.info(
+    `Uploading GitHub dependency snapshot from ${githubDependencySnapshotFile}`
+  );
+  debugLog("Snapshot:", snapshot);
+
+  const response = await client.postDependencySnapshot(snapshot);
```

Comment:
Do we need to handle errors for this call? Wondering about if we're not able to complete the API call for some reason